File: Sem1-2/CPP_Java2024/List3/CPP/src/Pentagon.cpp
#include <Pentagon.hpp>
#include <utils.hpp>

#include <stdexcept>
#include <cmath>

Pentagon::Pentagon(double side) : Shape("Pentagon"), side(side)
{
    if(side < 0)
    {
        throw std::invalid_argument(LOG_LOC() + "Pentagon side cannot be negative, got: " + std::to_string(side));
    }
}

double Pentagon::Circumference() const
{
    return 5.0 * side;
}

double Pentagon::Area() const
{
    return 1.25 * side * side * (1.0 / std::tan(36 * (M_PI / 180.0)));
}

File: Sem1-2/CPP_Java2024/List3/JAVA/Pentagon.java
public class Pentagon extends Shape
{
    protected final double side;

    public Pentagon(double side) throws IllegalArgumentException
    {
        super("Pentagon");
        if(side < 0)
        {
            throw new IllegalArgumentException("Pentagon side cannot be negative, got: " + side);
        }
        this.side = side;
    }

    @Override
    public double Circumference()
    {
        return 5.0 * side;
    }

    @Override
    public double Area()
    {
        return 1.25 * side * side * (1.0 / Math.tan(Math.toRadians(36)));
    }
}